scripts/sanity_chk/sanitylib.py

```diff
@@ -510,15 +511,15 @@ def device_is_available(self, instance):
         for i in self.suite.connected_hardware:
             if fixture and fixture not in i.get('fixtures', []):
                 continue
-            if i['platform'] == device and i['available'] and i['serial']:
+            if i['platform'] == device and i['available'] and (i['serial'] or i['serial_pty']):
```

Comment:
Perhaps the bit in () should move to a short function like device_is_valid() ?

Reply:
yeah, starting to dislike those long conditionals as they become unreadable, will see how it can be simplified.

Reply:
Maybe I name it as has_serial(), etc?